sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -0,0 +1,332 @@
+					  &hdmi[i]);
+		if (err < 0)
+			return err;
+
+		i++;
+	}
+
+	if (!component)
+		return -EINVAL;
+
+	return hdac_hdmi_jack_port_init(component, &card->dapm);
+}
+#else
+static int card_late_probe(struct snd_soc_card *card)
+{
+	return 0;
+}
+#endif
+
+static const struct snd_soc_dapm_widget widgets[] = {
+	SND_SOC_DAPM_HP("Headphones", NULL),
+	SND_SOC_DAPM_MIC("AMIC", NULL),
+	SND_SOC_DAPM_SPK("Speaker", NULL),
+};
+
+static const struct snd_soc_dapm_route map[] = {
+	/*Headphones*/
+	{ "Headphones", NULL, "HP" },
+	{ "MIC2", NULL, "AMIC" },
+};
+
+static const struct snd_kcontrol_new controls[] = {
+	SOC_DAPM_PIN_SWITCH("Headphones"),
+	SOC_DAPM_PIN_SWITCH("AMIC"),
+	SOC_DAPM_PIN_SWITCH("Speaker"),
+};
+
+static const struct snd_soc_dapm_route spk_rt1308_map[] = {
+	{ "Speaker", NULL, "rt1308-1 SPOL" },
+	{ "Speaker", NULL, "rt1308-1 SPOR" },
+	{ "Speaker", NULL, "rt1308-2 SPOL" },
+	{ "Speaker", NULL, "rt1308-2 SPOR" },
+};
+
+SND_SOC_DAILINK_DEF(sdw0_pin2,
+	DAILINK_COMP_ARRAY(COMP_CPU("SDW0 Pin2")));
+SND_SOC_DAILINK_DEF(sdw0_pin3,
+	DAILINK_COMP_ARRAY(COMP_CPU("SDW0 Pin3")));
+SND_SOC_DAILINK_DEF(sdw0_codec,
+	DAILINK_COMP_ARRAY(COMP_CODEC("sdw:0:25d:711:0:1", "rt711-aif1")));
+
+SND_SOC_DAILINK_DEF(sdw1_pin2,
+	DAILINK_COMP_ARRAY(COMP_CPU("SDW1 Pin2")));
+SND_SOC_DAILINK_DEF(sdw1_codec,
+	DAILINK_COMP_ARRAY(COMP_CODEC("sdw:1:25d:1308:0:0", "rt1308-aif")));
+
+SND_SOC_DAILINK_DEF(sdw2_pin2,
+	DAILINK_COMP_ARRAY(COMP_CPU("SDW2 Pin2")));
+SND_SOC_DAILINK_DEF(sdw2_codec,
+	DAILINK_COMP_ARRAY(COMP_CODEC("sdw:2:25d:1308:0:2", "rt1308-aif")));
+
+SND_SOC_DAILINK_DEF(sdw3_pin3,
+	DAILINK_COMP_ARRAY(COMP_CPU("SDW3 Pin3")));
+SND_SOC_DAILINK_DEF(sdw3_codec,
+	DAILINK_COMP_ARRAY(COMP_CODEC("sdw:3:25d:1308:0:1", "rt715-aif")));
+
+#if IS_ENABLED(CONFIG_SND_SOC_HDAC_HDMI)
+SND_SOC_DAILINK_DEF(idisp1_pin,
+	DAILINK_COMP_ARRAY(COMP_CPU("iDisp1 Pin")));
+SND_SOC_DAILINK_DEF(idisp1_codec,
+	DAILINK_COMP_ARRAY(COMP_CODEC("ehdaudio0D2", "intel-hdmi-hifi1")));
+
+SND_SOC_DAILINK_DEF(idisp2_pin,
+	DAILINK_COMP_ARRAY(COMP_CPU("iDisp2 Pin")));
+SND_SOC_DAILINK_DEF(idisp2_codec,
+	DAILINK_COMP_ARRAY(COMP_CODEC("ehdaudio0D2", "intel-hdmi-hifi2")));
+
+SND_SOC_DAILINK_DEF(idisp3_pin,
+	DAILINK_COMP_ARRAY(COMP_CPU("iDisp3 Pin")));
+SND_SOC_DAILINK_DEF(idisp3_codec,
+	DAILINK_COMP_ARRAY(COMP_CODEC("ehdaudio0D2", "intel-hdmi-hifi3")));
+#endif
+
+SND_SOC_DAILINK_DEF(platform,
+		DAILINK_COMP_ARRAY(COMP_PLATFORM("0000:00:1f.3")));
+
+static int rt1308_init(struct snd_soc_pcm_runtime *runtime)
+{
+	struct snd_soc_card *card = runtime->card;
+	int ret;
+
+	ret = snd_soc_dapm_add_routes(&card->dapm, spk_rt1308_map,
+				      ARRAY_SIZE(spk_rt1308_map));
+	if (ret)
+		dev_warn(card->dev, "add new routes failed %d\n", ret);
+
+	return ret;
+}
+
+static struct snd_soc_codec_conf rt1308_codec_conf[] = {
+	{
+		.dev_name = "sdw:1:25d:1308:0:0",
+		.name_prefix = "rt1308-1",
+	},
+	{
+		.dev_name = "sdw:2:25d:1308:0:2",
+		.name_prefix = "rt1308-2",
+	},
```

Comment:
We have to add a conf for rt715 as well since some control name are identical in rt715 and rt711 codec driver.